exporters/geneva/src/exporter.cc

```diff
@@ -0,0 +1,357 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+#include "opentelemetry/exporters/geneva/metrics/exporter.h"
+#include "opentelemetry/exporters/geneva/metrics/unix_domain_socket_data_transport.h"
+#include "opentelemetry/metrics/meter_provider.h"
+#include "opentelemetry/sdk_config.h"
+
+#include <memory>
+#include <mutex>
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace exporter {
+namespace geneva {
+namespace metrics {
+Exporter::Exporter(const ExporterOptions &options)
+    : options_(options), connection_string_parser_(options_.connection_string),
+      data_transport_{nullptr}, buffer_index_histogram_(0),
+      buffer_index_non_histogram_(0) {
+  if (connection_string_parser_.IsValid()) {
+    if (connection_string_parser_.transport_protocol_ ==
+        TransportProtocol::kUNIX) {
+      data_transport_ =
+          std::unique_ptr<DataTransport>(new UnixDomainSocketDataTransport(
+              connection_string_parser_.url_->path_));
+    }
+  }
+  // Connect transport at initialization
+  auto status = data_transport_->Connect();
+  if (!status) {
+    LOG_ERROR("Exporter::Exporter Connect failed. No data would be sent.");
```

Comment:
Return failure with error

Reply:
Good point. Logging error, and setting `is_shutdown_` flag to true in connect fails. This will ensure that subsequent Export() calls fail early.